libportal/notification.c

```diff
@@ -148,12 +497,66 @@ parse_notification (GTask *task)
             continue;
         }
 
-      g_variant_builder_add (data->builder, "{sv}", key, value);
+      if (strcmp (key, "icon") == 0)
+        {
+          /* This parse loop will be restarted in case of async loading */
+          if (parse_icon_property (task, value))
+            return;
+        }
+      else if (strcmp (key, "sound") == 0)
+        {
+          /* This parse loop will be restarted in case of async loading */
+          if (parse_sound_property (task, value))
+            return;
+        }
+      else if (portal->notification_interface_version < 2 && strcmp (key, "buttons") == 0)
```

Comment:
`>=2`, no?

Reply:
No, before version 2 we need to filter out properties version 1 doesn't support.